provision/aws/route53/route53_create_failover_lambda.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+set -e -o pipefail
+
+if [[ "$RUNNER_DEBUG" == "1" ]]; then
+  set -x
+fi
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source ${SCRIPT_DIR}/route53_common.sh
+
+if [ -z "${HEALTH_CHECK_ID}" ]; then
+  echo "HEALTH_CHECK_ID variable must be set"
+  exit 1
+fi
+
+ALARM_NAME=${HEALTH_CHECK_ID}
+TOPIC_NAME=${HEALTH_CHECK_ID}
+FUNCTION_NAME=${HEALTH_CHECK_ID}
+ROUTE53_REGION="us-east-1"
+
+export AWS_REGION=${ROUTE53_REGION}
+
+TOPIC_ARN=$(aws sns create-topic --name ${TOPIC_NAME} \
+  --query "TopicArn" \
+  --tags "Key=HealthCheckId,Value=${HEALTH_CHECK_ID}" \
+  --output text
+)
+
+# Create CloudWatch alarm
+aws cloudwatch put-metric-alarm \
```

Comment:
what is the outcome of this CloudWatch alarm action ? would it fail the HealthCheckStatus if it doesn't satisfy the threshold ?

Reply:
> what is the outcome of this CloudWatch alarm action ?

When the alarm is triggered a notification is sent to the SNS topic. A subscription is created on this topic for the AWS lambda, so when a notifcation is received the function is triggered.

> would it fail the HealthCheckStatus if it doesn't satisfy the threshold ?

The alarm is triggered if the `HealthCheckStatus` metric of the Route53 health check fails.

Reply:
great, thanks for the clarification Ryan.